askama/src/error.rs

```diff
@@ -71,6 +72,13 @@ impl From<::serde_json::Error> for Error {
     }
 }
 
+impl From<Infallible> for Error {
+    #[track_caller]
+    fn from(_: Infallible) -> Self {
+        unreachable!()
+    }
```

Comment:
```suggestion
    fn from(value: Infallible) -> Self {
        match value {}
    }
```

Alternatively use an empty `match`, as that results in _no_ generated code, whereas `unreachable!()` inserts panic related code